### ISSUE TITLE
Test set accuracy does not match with published result

### ISSUE THREAD

**Issue by cloudfengchao** [thunlp/OpenNRE#7]
As shown in the home page, test set accuracy should be around 0.8, while it is around 0.50 on my machine for all test tasks. I'm using tensorflow 1.0.0 and fixed some API migration issues before I could successfully train and test the model, and I was evaluating P@N for iter 14000. Below I pasted the training info and testing info. Please let me know where I could get it wrong. Thank you so much:)
============================================================
have saved model to ./model/ATT_GRU_model-13500
2017-04-02T03:08:14.022144: step 13550, softmax_loss 8.99307, acc 0.94
2017-04-02T03:08:24.764260: step 13600, softmax_loss 11.1679, acc 0.92
2017-04-02T03:08:35.354639: step 13650, softmax_loss 4.43873, acc 0.98
2017-04-02T03:08:45.710377: step 13700, softmax_loss 12.744, acc 0.94
2017-04-02T03:08:55.166245: step 13750, softmax_loss 2.57561, acc 0.98
2017-04-02T03:09:04.721789: step 13800, softmax_loss 2.85167, acc 0.98
2017-04-02T03:09:14.571793: step 13850, softmax_loss 2.9708, acc 0.98
2017-04-02T03:09:23.866147: step 13900, softmax_loss 4.50743, acc 0.98
2017-04-02T03:09:34.195987: step 13950, softmax_loss 5.79872, acc 0.96
2017-04-02T03:09:44.283621: step 14000, softmax_loss 10.5228, acc 0.92
saving model
have saved model to ./model/ATT_GRU_model-14000
out of range
2017-04-02T03:10:17.838734: step 14050, softmax_loss 2.30686, acc 0.98
2017-04-02T03:10:28.560130: step 14100, softmax_loss 5.12479, acc 0.98
2017-04-02T03:10:38.362117: step 14150, softmax_loss 8.63566, acc 0.94
2017-04-02T03:10:48.828123: stepI tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 56277731 get requests, put_count=56277740 evicted_count=17000 eviction_rate=0.000302073 and unsatisfied allocation rate=0.000307226
 14200, softmax_loss 5.62947, acc 0.98
2017-04-02T03:10:59.361231: step 14250, softmax_loss 4.52537, acc 0.98
2017-04-02T03:11:08.783026: step 14300, softmax_loss 32.7668, acc 0.84
2017-04-02T03:11:19.310379: step 14350, softmax_loss 10.8961, acc 0.94
2017-04-02T03:11:29.238714: step 14400, softmax_loss 7.69886, acc 0.94
2017-04-02T03:11:38.763790: step 14450, softmax_loss 14.5586, acc 0.92
2017-04-02T03:11:48.848076: step 14500, softmax_loss 14.7915, acc 0.96

 
============================================================

Evaluating P@N for iter 14000
Evaluating P@N for one
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 7082 get requests, put_count=1102 evicted_count=1000 eviction_rate=0.907441 and unsatisfied allocation rate=0.999718
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 100 to 110
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=6011 evicted_count=6000 eviction_rate=0.99817 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=4013 evicted_count=4000 eviction_rate=0.996761 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=3016 evicted_count=3000 eviction_rate=0.994695 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=1019 evicted_count=1000 eviction_rate=0.981354 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 4716 get requests, put_count=5254 evicted_count=5000 eviction_rate=0.951656 and unsatisfied allocation rate=0.950594
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 233 to 256
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=3025 evicted_count=3000 eviction_rate=0.991736 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=2030 evicted_count=2000 eviction_rate=0.985222 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=1037 evicted_count=1000 eviction_rate=0.96432 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 7082 get requests, put_count=7491 evicted_count=7000 eviction_rate=0.934455 and unsatisfied allocation rate=0.936317
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 449 to 493
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=6049 evicted_count=6000 eviction_rate=0.991899 and unsatisfied allocation rate=0
P@100:
0.53
P@200:
0.525
P@300:
0.51
Evaluating P@N for two
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=5059 evicted_count=5000 eviction_rate=0.988338 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=5072 evicted_count=5000 eviction_rate=0.985804 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=5087 evicted_count=5000 eviction_rate=0.982898 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=5105 evicted_count=5000 eviction_rate=0.979432 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 7082 get requests, put_count=6529 evicted_count=5000 eviction_rate=0.765814 and unsatisfied allocation rate=0.802033
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 1400 to 1540
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=1169 evicted_count=1000 eviction_rate=0.855432 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=1225 evicted_count=1000 eviction_rate=0.816327 and unsatisfied allocation rate=0
P@100:
0.56
P@200:
0.535
P@300:
0.503333333333
Evaluating P@N for all
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 7082 get requests, put_count=7271 evicted_count=4000 eviction_rate=0.550131 and unsatisfied allocation rate=0.576532
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 2997 to 3296
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 0 get requests, put_count=3398 evicted_count=3000 eviction_rate=0.882872 and unsatisfied allocation rate=0
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 11798 get requests, put_count=12147 evicted_count=4000 eviction_rate=0.329299 and unsatisfied allocation rate=0.350314
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 5305 to 5835
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 11798 get requests, put_count=11832 evicted_count=1000 eviction_rate=0.0845166 and unsatisfied allocation rate=0.141634
I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 7764 to 8540
P@100:
0.57
P@200:
0.535
P@300:
0.523333333333
2017-04-02T22:13:33.351736
Evaluating all test data and save data for PR curve
saving all test result...
PR curve area:0.257896324167
2017-04-02T22:14:57.170137
P@N for all test data:
P@100:
0.52
P@200:
0.505
P@300:
0.476666666667

**Comment by Tsingularity** [comment]
Have you evaluated the testing dataset using the model we provide in the ./model directory? If you then can get the high accuracy as we provided in Readme, it means there are nothing wrong with your test codes, which also means there are something wrong with your training settings, like learning rate .etc.

Please inform me if you still get any questions.

**Comment by cloudfengchao** [comment]
@Tsingularity 
Thank you for your prompt reply. I guess the problem might be that in train_GRU.py the training dataset is default to ./data/small_XXXX.npy, where I should replace with ./data/train_XXXX.npy.

**Comment by frankxu2004** [comment]
I had the same problem on this too.
I migrated the API for TensorFlow 1.0+ API and it turns out that the training process is very unstable and did not converge as expected:

2017-04-20T22:49:16.594582: step 50, softmax_loss 21.4735, acc 0.96
2017-04-20T22:49:32.141554: step 100, softmax_loss 15.4196, acc 0.94
2017-04-20T22:49:47.318107: step 150, softmax_loss 19.3769, acc 0.92
2017-04-20T22:50:02.992736: step 200, softmax_loss 11.1531, acc 0.98
2017-04-20T22:50:18.202700: step 250, softmax_loss 35.4513, acc 0.9
2017-04-20T22:50:32.981644: step 300, softmax_loss 6.89652, acc 0.98
2017-04-20T22:50:47.808317: step 350, softmax_loss 10.088, acc 0.96
2017-04-20T22:51:01.964026: step 400, softmax_loss 24.7675, acc 0.9
2017-04-20T22:51:17.774023: step 450, softmax_loss 14.7294, acc 0.94
2017-04-20T22:51:32.227491: step 500, softmax_loss 10.6867, acc 0.96
2017-04-20T22:51:48.735949: step 550, softmax_loss 36.2467, acc 0.86
2017-04-20T22:52:04.734398: step 600, softmax_loss 14.5625, acc 0.96
2017-04-20T22:52:20.174327: step 650, softmax_loss 24.6323, acc 0.9
2017-04-20T22:52:34.532755: step 700, softmax_loss 2.41186, acc 1
2017-04-20T22:52:49.000365: step 750, softmax_loss 13.4522, acc 0.94
2017-04-20T22:53:04.999005: step 800, softmax_loss 26.3403, acc 0.92
2017-04-20T22:53:19.506108: step 850, softmax_loss 24.053, acc 0.9
2017-04-20T22:53:34.691490: step 900, softmax_loss 18.0913, acc 0.92
2017-04-20T22:53:49.991637: step 950, softmax_loss 7.47336, acc 0.98
2017-04-20T22:54:05.118305: step 1000, softmax_loss 13.1802, acc 0.96
2017-04-20T22:54:19.439001: step 1050, softmax_loss 31.5119, acc 0.86
2017-04-20T22:54:34.327526: step 1100, softmax_loss 18.9508, acc 0.96
2017-04-20T22:54:48.353167: step 1150, softmax_loss 9.58851, acc 0.96
2017-04-20T22:55:03.408483: step 1200, softmax_loss 6.8638, acc 0.98
2017-04-20T22:55:19.836647: step 1250, softmax_loss 31.93, acc 0.88
2017-04-20T22:55:35.030056: step 1300, softmax_loss 5.76889, acc 0.98
2017-04-20T22:55:49.492384: step 1350, softmax_loss 17.2845, acc 0.96
2017-04-20T22:56:06.651158: step 1400, softmax_loss 14.9008, acc 0.94
2017-04-20T22:56:22.267086: step 1450, softmax_loss 23.0303, acc 0.9
2017-04-20T22:56:37.428491: step 1500, softmax_loss 15.9611, acc 0.94
2017-04-20T22:56:53.297016: step 1550, softmax_loss 10.245, acc 0.96
2017-04-20T22:57:07.970155: step 1600, softmax_loss 23.6018, acc 0.9
2017-04-20T22:57:25.765035: step 1650, softmax_loss 7.96503, acc 0.98
2017-04-20T22:57:42.618245: step 1700, softmax_loss 16.4669, acc 0.94
2017-04-20T22:57:57.278566: step 1750, softmax_loss 18.9911, acc 0.94
2017-04-20T22:58:12.971772: step 1800, softmax_loss 9.03051, acc 0.96

and for testing using the exactly same code, the same situation occurs like @cloudfengchao 

As for using the provided Model file, it is unfortunate that the RNN cells got renamed inside tf.contribs, thus rendering pre-trained model files incompatible across 0.x Tensorflow and 1.x Tensorflow.

Any help would be really appreciated!

**Comment by sousuodaren** [comment]
@cloudfengchao @frankxu2004 Did you get the result as published result? I have no idea about the problem.

**Comment by frankxu2004** [comment]
@sousuodaren you can check my updated fork (https://github.com/frankxu2004/TensorFlow-NRE) that is upgraded for Python 3.x and Tensorflow 1.x version. I have added a compatibility layer so that the provided original pre-trained model file could be now correctly read and used. With that model file, I was able to reproduce the test result (with acceptable error) described on my fork.

However, I am still not sure about the training part. I was not able to reproduce the model trained using the provided code and data. I'm suspecting that these small training data might have caused it:
```
    train_y = np.load('./data/small_y.npy')
    train_word = np.load('./data/small_word.npy')
    train_pos1 = np.load('./data/small_pos1.npy')
    train_pos2 = np.load('./data/small_pos2.npy')

```

**Comment by pfllo** [comment]
@frankxu2004  I have tried replacing files like "small_word.npy" with "train_word.npy", but still cannot reproduce the result reported in README, with the default hyper-parameters of this repository. But the PR curve is comparable to PCNN+ATT.

@Tsingularity Did you generate the results in README with the hyper-parameters in this repository, or with another set of hyper-parameters?  If with another set of hyper-parameters, could you give us the detailed settings?  Thanks a lot.

**Comment by mbxmxm** [comment]
@Tsingularity 我把您的代码放到python 3.x，tensorflow 1.x的环境下运行，修改之后没有报错，只是存在两个问题：1、训练的时候softmax_loss变得极低，趋近于0；2、用这个训练结果测试，P@N输出都是0.0。我在tenorflow 0.12环境下是没有问题的，只是在tensorflow 1.x环境下出现这样的问题。我输出了测试的中间数据，对比了一下两个环境下除了参数order在tensorflow 1.x下总是以0开始，其他参数并无明显不同，想请问一下这是什么原因？谢谢

**Comment by pvthuy** [comment]
I couldn't reproduce the results of `BGRU+2ATT` shown in this repository.
`PCNN+ATT` is still better a little than `BGRU+2ATT` after I run the code.

Did anyone have similar results?

**Comment by mbxmxm** [comment]
For Tensorflow 1.x, I use the same code with you, but there was a problem that the GPU memory is not enough, but my GPU memory is 8G. and no other program used GPU. Have you meet this problem. As for the result, I can get the better result but when I use the plot_pr.py to draw the PR curve, there is no line for the BGRU+2ATT, I don't know why.

**Comment by pvthuy** [comment]
@mbxmxm I used the GPU Tesla K40 with 11GB but the code run too slow so I terminated it after half a day (only 2 epochs were running). Actually, I do not know the reason.

BTW, how did you know that you got the better result? What is the exact performance?
For plotting the PR curves, please change the model name in file `plot_pr.py`.

**Comment by mbxmxm** [comment]
@pvthuy Thanks, as for the result,I have print the precision and recall,and I just ran the program with Tensorflow r0.12 once.

**Comment by pvthuy** [comment]
@mbxmxm After plotting the PR curves, could you check if `BGRU+2ATT` is better than `PCNN+ATT` or not? Thanks!

**Comment by mbxmxm** [comment]
@pvthuy I have tried, the result depends on which model you choose. actually, most models performed worse than PCNN-ATT, but some mode perform better

**Comment by pvthuy** [comment]
@mbxmxm Can you show the comparison between 2 models, in the case `BGRU+2ATT` is better?

**Comment by mbxmxm** [comment]
@pvthuy How should I show you the result? It's a picture. Can you give me your email?

**Comment by pvthuy** [comment]
@mbxmxm What is the best area under PR curves of `BGRU+ATT` you got? E.g., 0.37?

**Comment by mbxmxm** [comment]
@pvthuy  there are two result: 0.341 and 0.347
